backend-shared/config/db/guardrow_test.go

```diff
@@ -0,0 +1,890 @@
+				DBRelationType:       db.APICRToDatabaseMapping_DBRelationType_SyncOperation,
+				DBRelationKey:        "test-key",
+			}
+
+			err = dbq.CreateAPICRToDatabaseMapping(ctx, &Apicrtodatabasemappingfirst)
+			Expect(err).To(BeNil())
+
+			Apicrtodatabasemappingsecond := db.APICRToDatabaseMapping{
+				APIResourceType:      "test-GitOpsDeployment",
+				APIResourceUID:       "test-k8s-uid-second",
+				APIResourceName:      "test-k8s-name",
+				APIResourceNamespace: "test-k8s-namespace",
+				NamespaceUID:         "test-namespace-uid",
+				DBRelationType:       "test-sync-operation",
+				DBRelationKey:        "test-key-second",
+			}
+			err = dbq.CreateAPICRToDatabaseMapping(ctx, &Apicrtodatabasemappingsecond)
+			Expect(err).To(BeNil())
+
+			rowsAffected, err := dbq.DeleteAPICRToDatabaseMapping(ctx, &Apicrtodatabasemappingsecond)
+			Expect(err).To(BeNil())
+			Expect(rowsAffected).To(Equal((1)))
+
+			err = dbq.GetDatabaseMappingForAPICR(ctx, &Apicrtodatabasemappingfirst)
+			Expect(err).To(BeNil())
+
+			err = dbq.GetDatabaseMappingForAPICR(ctx, &Apicrtodatabasemappingsecond)
+			Expect(true).To(Equal(db.IsResultNotFoundError(err)))
+
+		})
+
+		It("Should test guard row against update and delete on application", func() {
+			err := db.SetupForTestingDBGinkgo()
+			Expect(err).To(BeNil())
+
+			ctx := context.Background()
+			dbq, err := db.NewUnsafePostgresDBQueries(true, true)
+			Expect(err).To(BeNil())
+			defer dbq.CloseDatabase()
+
+			_, managedEnvironment, _, gitopsEngineInstance, _, err := db.CreateSampleData(dbq)
+			Expect(err).To(BeNil())
+
+			applicationfirst := db.Application{
+				Application_id:          "test-my-application-1",
+				Name:                    "my-application",
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+
+			err = dbq.CreateApplication(ctx, &applicationfirst)
+			Expect(err).To(BeNil())
+
+			applicationsecond := db.Application{
+				Application_id:          "test-my-application-2",
+				Name:                    "my-application",
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+			err = dbq.CreateApplication(ctx, &applicationsecond)
+			Expect(err).To(BeNil())
+
+			applicationsecond = db.Application{
+				Application_id:          applicationsecond.Application_id,
+				Name:                    "test-application-update",
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+				SeqID:                   applicationsecond.SeqID,
+			}
+
+			err = dbq.UpdateApplication(ctx, &applicationsecond)
+			Expect(err).To(BeNil())
+			Expect(applicationsecond.Name).Should(Equal("test-application-update"))
```

Comment:
Need to do a `dbq.GetApplication` for both first and second here, to ensure they have been updated with the new state of the database: as the Go objects won't be updated automatically (by the go-pg library we use).